### ISSUE TITLE
False positives by 'Lint/AmbiguousOperator' cop

### ISSUE THREAD

**Issue by agalloch** [bbatsov/rubocop#5476]
Hello, guys, I'm not sure this is an issue or a Rubocop quirk so decided to ask 🙂 

Using recent version of RSpec and Rubocop version (below), `Lint/AmbiguousOperator` complains and auto-corrects like this:
```diff
- expect('foo').to eq -1.23
+ expect('foo').to eq(-1.23)
```

I understand there are cases when calling `eq` without braces is ambiguous but how come a Float literal argument be ambiguous? Plus, [Ruby style guide says](https://github.com/bbatsov/ruby-style-guide#method-invocation-parens):
> Always omit parentheses for:
> - Methods that are part of an internal DSL (e.g., Rake, Rails, **_RSpec_**):
> ```
> # bad
> validates(:name, presence: true)
> # good
> validates :name, presence: true
> ```

## Expected behavior
Rubocop should not complain

## Actual behavior
Rubocop complains

## RuboCop version
```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by mikegee** [comment]
This cop is surfacing a Ruby warning:

```
% ruby -W3 -e"expect('foo').to eq -1.23"
-e:1: warning: ambiguous first argument; put parentheses or a space even after `-' operator
```

**Comment by agalloch** [comment]
@mikegee, thank you for the quick response! 🙂

Isn't there a contradiction then between the Ruby style guide (as quoted about internal DSLs) and `Lint/AmbiguousOperator`? Why does the cop not auto-correct like, e.g.:
```diff
- expect('foo').to eq -1.23
+ expect('foo').to(eq(-1.23))    # <- mind the extra braces
```
what am I missing?

**Comment by mikegee** [comment]
The parens for `to()` don't disambiguate the `-`. The ambiguity in `expect('foo').to eq -1.23` is only between `expect('foo').to eq(-1.23)` and `expect('foo').to eq() - 1.23`.

**Comment by agalloch** [comment]
OK, thanks for your time!

**Comment by mikegee** [comment]
You’re welcome.
